lib/improver/tests/nowcasting/accumulation/test_Accumulation.py

```diff
@@ -0,0 +1,421 @@
+        """Test the default accumulation_units are set when not specified."""
+        plugin = Accumulation()
+        self.assertEqual(plugin.accumulation_units, "m")
+        self.assertEqual(plugin.accumulation_period, None)
+
+    def test_units_set(self):
+        """Test the accumulation_units are set when specified."""
+        plugin = Accumulation(accumulation_units="cm")
+        self.assertEqual(plugin.accumulation_units, "cm")
+
+    def test_period_set(self):
+        """Test the accumulation_period is set when specified."""
+        plugin = Accumulation(accumulation_period=180)
+        self.assertEqual(plugin.accumulation_period, 180)
+
+
+class Test__repr__(IrisTest):
+    """Test class representation"""
+
+    def test_basic(self):
+        """Test string representation"""
+        result = str(Accumulation(accumulation_units="cm",
+                                  accumulation_period=60))
+        expected_result = ("<Accumulation: accumulation_units=cm, "
+                           "accumulation_period=60>")
+        self.assertEqual(result, expected_result)
+
+
+class Test_sort_cubes_by_time(rate_cube_set_up):
+    """Tests the input cubes are sorted in time ascending order."""
+
+    def test_returns_cubelist(self):
+        """Test function returns a cubelist."""
+
+        result, _ = Accumulation.sort_cubes_by_time(self.cubes)
+        self.assertIsInstance(result, iris.cube.CubeList)
+
+    def test_reorders(self):
+        """Test function reorders a cubelist that is not time ordered."""
+
+        expected = [cube.coord('time').points[0] for cube in self.cubes]
+        self.cubes = self.cubes[::-1]
+        reordered = [cube.coord('time').points[0] for cube in self.cubes]
+
+        result, _ = Accumulation.sort_cubes_by_time(self.cubes)
+        result_times = [cube.coord('time').points[0] for cube in result]
+
+        self.assertIsInstance(result, iris.cube.CubeList)
+        self.assertEqual(result_times, expected)
+        self.assertNotEqual(result_times, reordered)
+
+    def test_times(self):
+        """Test function returns the correct times for the sorted cubes."""
+
+        expected = [1510286400, 1510286460, 1510286520, 1510286580,
+                    1510286640, 1510286700, 1510286760, 1510286820,
+                    1510286880, 1510286940, 1510287000]
+        _, times = Accumulation.sort_cubes_by_time(self.cubes)
+
+        self.assertArrayEqual(times, expected)
+
+
+class Test_get_period_sets(rate_cube_set_up):
+    """Tests the requested accumulation_period can be constructed from the
+    input cubes and return lists of the required inputs to construct each
+    such period."""
+
+    def test_returns_correct_type(self):
+        """Test function returns the expected list."""
+
+        time_interval = 60
+        plugin = Accumulation(accumulation_period=120)
+        result = plugin.get_period_sets(time_interval, self.cubes)
+        self.assertIsInstance(result, list)
```

Comment:
There's no distinction between this list and the list returned by the next test.  Could we test the values here (or add another test) to show that a complete set of inputs with an accumulation period is correctly split up?

Reply:
I missed that completely. I've added an extra test.